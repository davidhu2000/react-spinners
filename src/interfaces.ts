import { SerializedStyles } from '@emotion/react';

export interface LengthObject {
  value: number;
  unit: string;
}

interface CommonProps {
  color?: string;
  loading?: boolean;
  css?: string | SerializedStyles;
}

export type LengthType = number | string;

export interface LoaderHeightWidthProps extends CommonProps {
  height?: LengthType;
  width?: LengthType;
}

export interface LoaderSizeProps extends CommonProps {
  size?: LengthType;
}

export interface LoaderSizeMarginProps extends LoaderSizeProps {
  margin?: LengthType;
}

export interface LoaderHeightWidthRadiusProps extends LoaderHeightWidthProps {
  margin?: LengthType;
  radius?: LengthType;
}
