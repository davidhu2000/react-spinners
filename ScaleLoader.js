(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['exports', 'emotion', 'react', 'prop-types', '@emotion/core', 'recompose/onlyUpdateForKeys'], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require('emotion'), require('react'), require('prop-types'), require('@emotion/core'), require('recompose/onlyUpdateForKeys'));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.emotion, global.react, global.propTypes, global.core, global.onlyUpdateForKeys);
    global.ScaleLoader = mod.exports;
  }
})(this, function (exports, _emotion, _react, _propTypes, _core, _onlyUpdateForKeys) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });

  var _react2 = _interopRequireDefault(_react);

  var _propTypes2 = _interopRequireDefault(_propTypes);

  var _onlyUpdateForKeys2 = _interopRequireDefault(_onlyUpdateForKeys);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  function _classCallCheck(instance, Constructor) {
    if (!(instance instanceof Constructor)) {
      throw new TypeError("Cannot call a class as a function");
    }
  }

  var _createClass = function () {
    function defineProperties(target, props) {
      for (var i = 0; i < props.length; i++) {
        var descriptor = props[i];
        descriptor.enumerable = descriptor.enumerable || false;
        descriptor.configurable = true;
        if ("value" in descriptor) descriptor.writable = true;
        Object.defineProperty(target, descriptor.key, descriptor);
      }
    }

    return function (Constructor, protoProps, staticProps) {
      if (protoProps) defineProperties(Constructor.prototype, protoProps);
      if (staticProps) defineProperties(Constructor, staticProps);
      return Constructor;
    };
  }();

  function _possibleConstructorReturn(self, call) {
    if (!self) {
      throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
    }

    return call && (typeof call === "object" || typeof call === "function") ? call : self;
  }

  function _inherits(subClass, superClass) {
    if (typeof superClass !== "function" && superClass !== null) {
      throw new TypeError("Super expression must either be null or a function, not " + typeof superClass);
    }

    subClass.prototype = Object.create(superClass && superClass.prototype, {
      constructor: {
        value: subClass,
        enumerable: false,
        writable: true,
        configurable: true
      }
    });
    if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;
  }

  var scale = (0, _core.keyframes)('0%{transform:scaley(1.0)}50%{transform:scaley(0.4)}100%{transform:scaley(1.0)}');

  var Loader = function (_React$Component) {
    _inherits(Loader, _React$Component);

    function Loader() {
      var _ref;

      var _temp, _this, _ret;

      _classCallCheck(this, Loader);

      for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
        args[_key] = arguments[_key];
      }

      return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_ref = Loader.__proto__ || Object.getPrototypeOf(Loader)).call.apply(_ref, [this].concat(args))), _this), _this.style = function (i) {
        var _this$props = _this.props,
            color = _this$props.color,
            width = _this$props.width,
            height = _this$props.height,
            margin = _this$props.margin,
            radius = _this$props.radius,
            widthUnit = _this$props.widthUnit,
            heightUnit = _this$props.heightUnit,
            radiusUnit = _this$props.radiusUnit;


        return (0, _core.css)('{background-color:', color, ';width:', '' + width + widthUnit, ';height:', '' + height + heightUnit, ';margin:', margin, ';border-radius:', '' + radius + radiusUnit, ';display:inline-block;animation:', scale, ' 1s ', i * 0.1, 's infinite cubic-bezier(.2,.68,.18,1.08);animation-fill-mode:both;}');
      }, _this.wrapper = function () {
        return _this.props.css || '';
      }, _temp), _possibleConstructorReturn(_this, _ret);
    }

    _createClass(Loader, [{
      key: 'render',
      value: function render() {
        var loading = this.props.loading;


        return loading ? _react2.default.createElement(
          'div',
          {
            className: (0, _emotion.css)(this.wrapper())
          },
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.style(1))
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.style(2))
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.style(3))
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.style(4))
          }),
          _react2.default.createElement('div', {
            className: (0, _emotion.css)(this.style(5))
          })
        ) : null;
      }
    }]);

    return Loader;
  }(_react2.default.Component);

  Loader.propTypes = {
    loading: _propTypes2.default.bool,
    color: _propTypes2.default.string,
    height: _propTypes2.default.number,
    width: _propTypes2.default.number,
    margin: _propTypes2.default.string,
    radius: _propTypes2.default.number,
    heightUnit: _propTypes2.default.string,
    widthUnit: _propTypes2.default.string,
    radiusUnit: _propTypes2.default.string,
    css: _propTypes2.default.string
  };

  Loader.defaultProps = {
    loading: true,
    color: '#000000',
    height: 35,
    width: 4,
    margin: '2px',
    radius: 2,
    heightUnit: 'px',
    widthUnit: 'px',
    radiusUnit: 'px',
    css: ''
  };

  var Component = (0, _onlyUpdateForKeys2.default)(['loading', 'color', 'height', 'width', 'margin', 'radius', 'heightUnit', 'widthUnit', 'radiusUnit', 'css'])(Loader);
  Component.defaultProps = Loader.defaultProps;
  exports.default = Component;
});