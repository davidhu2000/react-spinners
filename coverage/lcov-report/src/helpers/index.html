
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/helpers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/helpers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>86/86</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.88% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>32/36</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>9/9</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>85/85</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="animation.ts"><a href="animation.ts.html">animation.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file high" data-value="colors.ts"><a href="colors.ts.html">colors.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="47" class="abs high">47/47</td>
	<td data-value="95" class="pct high">95%</td>
	<td data-value="20" class="abs high">19/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="46" class="abs high">46/46</td>
	</tr>

<tr>
	<td class="file high" data-value="unitConverter.ts"><a href="unitConverter.ts.html">unitConverter.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Jun 25 2024 21:55:12 GMT-0700 (Pacific Daylight Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    