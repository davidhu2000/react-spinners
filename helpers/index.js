(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(['exports'], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports);
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports);
    global.index = mod.exports;
  }
})(this, function (exports) {
  'use strict';

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  var calculateRgba = exports.calculateRgba = function calculateRgba(color, opacity) {
    if (color[0] === '#') {
      color = color.slice(1);
    }

    if (color.length === 3) {
      var res = '';
      color.split('').forEach(function (c) {
        res += c;
        res += c;
      });
      color = res;
    }

    var rgbValues = color.match(/.{2}/g).map(function (hex) {
      return parseInt(hex, 16);
    }).join(', ');
    return 'rgba(' + rgbValues + ', ' + opacity + ')';
  };
});