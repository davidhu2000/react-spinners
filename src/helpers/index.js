export const calculateRgba = (color, opacity) => {
  if (color[0] === '#') {
    color = color.slice(1);
  }

  if (color.length === 3) {
    let res = '';
    color.split('').forEach(c => {
      res += c;
      res += c;
    });
    color = res;
  }

  let rgbValues = color.match(/.{2}/g).map(hex => parseInt(hex, 16)).join(', ');
  return `rgba(${rgbValues}, ${opacity})`;
};

export * from './proptypes';
