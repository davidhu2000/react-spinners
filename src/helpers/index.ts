export * from "./proptypes";
export * from "./colors";
export * from "./unitConverter";
