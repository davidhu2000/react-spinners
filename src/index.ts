import BarLoaderComponent from "./BarLoader";
import BeatLoaderComponent from "./BeatLoader";
import BounceLoaderComponent from "./BounceLoader";
import CircleLoaderComponent from "./CircleLoader";
import ClimbingBoxLoaderComponent from "./ClimbingBoxLoader";
import ClipLoaderComponent from "./ClipLoader";
import ClockLoaderComponent from "./ClockLoader";
import DotLoaderComponent from "./DotLoader";
import FadeLoaderComponent from "./FadeLoader";
import GridLoaderComponent from "./GridLoader";
import HashLoaderComponent from "./HashLoader";
import MoonLoaderComponent from "./MoonLoader";
import PacmanLoaderComponent from "./PacmanLoader";
import PropagateLoaderComponent from "./PropagateLoader";
import PulseLoaderComponent from "./PulseLoader";
import PuffLoaderComponent from "./PuffLoader";
import RingLoaderComponent from "./RingLoader";
import RiseLoaderComponent from "./RiseLoader";
import RotateLoaderComponent from "./RotateLoader";
import ScaleLoaderComponent from "./ScaleLoader";
import SkewLoaderComponent from "./SkewLoader";
import SquareLoaderComponent from "./SquareLoader";
import SyncLoaderComponent from "./SyncLoader";

import {
  LoaderHeightWidthProps,
  LoaderSizeMarginProps,
  LoaderSizeProps,
  LoaderHeightWidthRadiusProps
} from "./interfaces";

export const BarLoader: React.ComponentClass<LoaderHeightWidthProps> = BarLoaderComponent;
export const BeatLoader: React.ComponentClass<LoaderSizeMarginProps> = BeatLoaderComponent;
export const BounceLoader: React.ComponentClass<LoaderSizeProps> = BounceLoaderComponent;
export const CircleLoader: React.ComponentClass<LoaderSizeProps> = CircleLoaderComponent;
export const ClimbingBoxLoader: React.ComponentClass<LoaderSizeProps> = ClimbingBoxLoaderComponent;
export const ClipLoader: React.ComponentClass<LoaderSizeProps> = ClipLoaderComponent;
export const ClockLoader: React.ComponentClass<LoaderSizeProps> = ClockLoaderComponent;
export const DotLoader: React.ComponentClass<LoaderSizeProps> = DotLoaderComponent;
export const FadeLoader: React.ComponentClass<LoaderHeightWidthRadiusProps> = FadeLoaderComponent;
export const GridLoader: React.ComponentClass<LoaderSizeMarginProps> = GridLoaderComponent;
export const HashLoader: React.ComponentClass<LoaderSizeProps> = HashLoaderComponent;
export const MoonLoader: React.ComponentClass<LoaderSizeProps> = MoonLoaderComponent;
export const PacmanLoader: React.ComponentClass<LoaderSizeMarginProps> = PacmanLoaderComponent;
export const PropagateLoader: React.ComponentClass<LoaderSizeProps> = PropagateLoaderComponent;
export const PulseLoader: React.ComponentClass<LoaderSizeMarginProps> = PulseLoaderComponent;
export const PuffLoader: React.ComponentClass<LoaderSizeProps> = PuffLoaderComponent;
export const RingLoader: React.ComponentClass<LoaderSizeProps> = RingLoaderComponent;
export const RiseLoader: React.ComponentClass<LoaderSizeMarginProps> = RiseLoaderComponent;
export const RotateLoader: React.ComponentClass<LoaderSizeMarginProps> = RotateLoaderComponent;
export const ScaleLoader: React.ComponentClass<LoaderHeightWidthRadiusProps> = ScaleLoaderComponent;
export const SkewLoader: React.ComponentClass<LoaderSizeProps> = SkewLoaderComponent;
export const SquareLoader: React.ComponentClass<LoaderSizeProps> = SquareLoaderComponent;
export const SyncLoader: React.ComponentClass<LoaderSizeProps> = SyncLoaderComponent;
