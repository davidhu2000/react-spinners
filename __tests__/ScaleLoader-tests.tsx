import * as React from "react";
import { mount, ReactWrapper } from "enzyme";
import { matchers } from "jest-emotion";
expect.extend(matchers);

import ScaleLoader from "../src/ScaleLoader";
import { LoaderHeightWidthRadiusProps } from "../src/interfaces";
import { heightWidthRadiusDefaults } from "../src/helpers";

describe("ScaleLoader", () => {
  let loader: ReactWrapper;
  let props: LoaderHeightWidthRadiusProps;
  let defaultColor: string = "#000000";
  let defaultHeight: number = 35;
  let defaultWidth: number = 4;
  let defaultRadius: number = 2;
  let defaultMargin: number = 2;
  let defaultUnit: string = "px";

  it("should match snapshot", () => {
    loader = mount(<ScaleLoader />);
    expect(loader).toMatchSnapshot();
  });

  it("should contain default props if no props are passed", () => {
    props = loader.props();
    expect(props).toEqual(heightWidthRadiusDefaults(defaultHeight, defaultWidth, defaultRadius));
  });

  it("should contain styles created using default props", () => {
    for (let i: number = 0; i < 5; i++) {
      expect(loader.find("div div").at(i)).toHaveStyleRule("background-color", defaultColor);
      expect(loader.find("div div").at(i)).toHaveStyleRule(
        "height",
        `${defaultHeight}${defaultUnit}`
      );
      expect(loader.find("div div").at(i)).toHaveStyleRule(
        "width",
        `${defaultWidth}${defaultUnit}`
      );
      expect(loader.find("div div").at(i)).toHaveStyleRule(
        "border-radius",
        `${defaultRadius}${defaultUnit}`
      );
      expect(loader.find("div div").at(i)).toHaveStyleRule(
        "margin",
        `${defaultMargin}${defaultUnit}`
      );
    }
  });

  it("should render null if loading prop is set as false", () => {
    loader = mount(<ScaleLoader loading={false} />);
    expect(loader.isEmptyRender()).toBe(true);
  });

  it("should render the correct color based on prop", () => {
    let color: string = "#e2e2e2";
    loader = mount(<ScaleLoader color={color} />);
    for (let i: number = 0; i < 5; i++) {
      expect(loader.find("div div")).not.toHaveStyleRule("background-color", defaultColor);
      expect(loader.find("div div")).toHaveStyleRule("background-color", color);
    }
  });

  describe("height props", () => {
    it("should render the height with px unit when height is a number", () => {
      let height: number = 18;
      loader = mount(<ScaleLoader height={height} />);

      expect(loader.find("div div")).not.toHaveStyleRule(
        "height",
        `${defaultHeight}${defaultUnit}`
      );
      expect(loader.find("div div")).toHaveStyleRule("height", `${height}${defaultUnit}`);
    });

    it("should render the height as is when height is a string with valid css unit", () => {
      let height: string = "18px";
      loader = mount(<ScaleLoader height={height} />);

      expect(loader.find("div div")).not.toHaveStyleRule(
        "height",
        `${defaultHeight}${defaultUnit}`
      );
      expect(loader.find("div div")).toHaveStyleRule("height", `${height}`);
    });

    it("should render the height with default unit of px when the unit is incorrect", () => {
      let length: number = 18;
      let unit: string = "ad";
      let height: string = `${length}${unit}`;
      loader = mount(<ScaleLoader height={height} />);

      expect(loader.find("div div")).not.toHaveStyleRule(
        "height",
        `${defaultHeight}${defaultUnit}`
      );
      expect(loader.find("div div")).toHaveStyleRule("height", `${length}${defaultUnit}`);
    });
  });

  describe("width props", () => {
    it("should render the width with px unit when width is a number", () => {
      let width: number = 18;
      loader = mount(<ScaleLoader width={width} />);

      expect(loader.find("div div")).not.toHaveStyleRule("width", `${defaultWidth}${defaultUnit}`);
      expect(loader.find("div div")).toHaveStyleRule("width", `${width}${defaultUnit}`);
    });

    it("should render the width as is when width is a string with valid css unit", () => {
      let width: string = "18px";
      loader = mount(<ScaleLoader width={width} />);

      expect(loader.find("div div")).not.toHaveStyleRule("width", `${defaultWidth}${defaultUnit}`);
      expect(loader.find("div div")).toHaveStyleRule("width", `${width}`);
    });

    it("should render the width with default unit of px when the unit is incorrect", () => {
      let length: number = 18;
      let unit: string = "ad";
      let width: string = `${length}${unit}`;
      loader = mount(<ScaleLoader width={width} />);

      expect(loader.find("div div")).not.toHaveStyleRule("width", `${defaultWidth}${defaultUnit}`);
      expect(loader.find("div div")).toHaveStyleRule("width", `${length}${defaultUnit}`);
    });
  });

  describe("radius props", () => {
    it("should render the radius with px unit when radius is a number", () => {
      let radius: number = 18;
      loader = mount(<ScaleLoader radius={radius} />);

      expect(loader.find("div div")).not.toHaveStyleRule(
        "border-radius",
        `${defaultRadius}${defaultUnit}`
      );
      expect(loader.find("div div")).toHaveStyleRule("border-radius", `${radius}${defaultUnit}`);
    });

    it("should render the radius as is when radius is a string with valid css unit", () => {
      let radius: string = "18px";
      loader = mount(<ScaleLoader radius={radius} />);

      expect(loader.find("div div")).not.toHaveStyleRule(
        "border-radius",
        `${defaultRadius}${defaultUnit}`
      );
      expect(loader.find("div div")).toHaveStyleRule("border-radius", `${radius}`);
    });

    it("should render the radius with default unit of px when the unit is incorrect", () => {
      let length: number = 18;
      let unit: string = "ad";
      let radius: string = `${length}${unit}`;
      loader = mount(<ScaleLoader radius={radius} />);

      expect(loader.find("div div")).not.toHaveStyleRule(
        "border-radius",
        `${defaultRadius}${defaultUnit}`
      );
      expect(loader.find("div div")).toHaveStyleRule("border-radius", `${length}${defaultUnit}`);
    });
  });

  describe("margin props", () => {
    it("should render the margin with px unit when margin is a number", () => {
      let margin: number = 18;
      loader = mount(<ScaleLoader margin={margin} />);

      for (let i: number = 0; i < 5; i++) {
        expect(loader.find("div div").at(i)).not.toHaveStyleRule(
          "margin",
          `${defaultMargin}${defaultUnit}`
        );
        expect(loader.find("div div").at(i)).toHaveStyleRule("margin", `${margin}${defaultUnit}`);
      }
    });

    it("should render the margin as is when margin is a string with valid css unit", () => {
      let margin: string = "18px";
      loader = mount(<ScaleLoader margin={margin} />);

      for (let i: number = 0; i < 5; i++) {
        expect(loader.find("div div").at(i)).not.toHaveStyleRule(
          "margin",
          `${defaultMargin}${defaultUnit}`
        );
        expect(loader.find("div div").at(i)).toHaveStyleRule("margin", `${margin}`);
      }
    });

    it("should render the margin with default unit of px when the unit is incorrect", () => {
      let length: number = 18;
      let unit: string = "ad";
      let margin: string = `${length}${unit}`;
      loader = mount(<ScaleLoader margin={margin} />);

      for (let i: number = 0; i < 5; i++) {
        expect(loader.find("div div").at(i)).not.toHaveStyleRule(
          "margin",
          `${defaultMargin}${defaultUnit}`
        );
        expect(loader.find("div div").at(i)).toHaveStyleRule("margin", `${length}${defaultUnit}`);
      }
    });
  });

  it("should render the css override based on props", () => {
    loader = mount(
      <ScaleLoader css={"position: absolute; width: 100px; height: 200px; color: blue;"} />
    );
    expect(loader).toHaveStyleRule("position", "absolute");
    expect(loader).toHaveStyleRule("width", "100px");
    expect(loader).toHaveStyleRule("height", "200px");
    expect(loader).toHaveStyleRule("color", "blue");
  });
});
