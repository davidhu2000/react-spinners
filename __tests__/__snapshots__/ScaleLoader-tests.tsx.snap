// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScaleLoader should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }
}

@keyframes animation-0 {
  0% {
    -webkit-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }

  50% {
    -webkit-transform: scaley(0.4);
    -ms-transform: scaley(0.4);
    transform: scaley(0.4);
  }

  100% {
    -webkit-transform: scaley(1.0);
    -ms-transform: scaley(1.0);
    transform: scaley(1.0);
  }
}

.emotion-0 {
  background-color: #000000;
  width: 4px;
  height: 35px;
  margin: 2px;
  border-radius: 2px;
  display: inline-block;
  -webkit-animation: animation-0 1s 0.1s infinite cubic-bezier(0.2,0.68,0.18,1.08);
  animation: animation-0 1s 0.1s infinite cubic-bezier(0.2,0.68,0.18,1.08);
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.emotion-1 {
  background-color: #000000;
  width: 4px;
  height: 35px;
  margin: 2px;
  border-radius: 2px;
  display: inline-block;
  -webkit-animation: animation-0 1s 0.2s infinite cubic-bezier(0.2,0.68,0.18,1.08);
  animation: animation-0 1s 0.2s infinite cubic-bezier(0.2,0.68,0.18,1.08);
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.emotion-2 {
  background-color: #000000;
  width: 4px;
  height: 35px;
  margin: 2px;
  border-radius: 2px;
  display: inline-block;
  -webkit-animation: animation-0 1s 0.30000000000000004s infinite cubic-bezier(0.2,0.68,0.18,1.08);
  animation: animation-0 1s 0.30000000000000004s infinite cubic-bezier(0.2,0.68,0.18,1.08);
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.emotion-3 {
  background-color: #000000;
  width: 4px;
  height: 35px;
  margin: 2px;
  border-radius: 2px;
  display: inline-block;
  -webkit-animation: animation-0 1s 0.4s infinite cubic-bezier(0.2,0.68,0.18,1.08);
  animation: animation-0 1s 0.4s infinite cubic-bezier(0.2,0.68,0.18,1.08);
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

.emotion-4 {
  background-color: #000000;
  width: 4px;
  height: 35px;
  margin: 2px;
  border-radius: 2px;
  display: inline-block;
  -webkit-animation: animation-0 1s 0.5s infinite cubic-bezier(0.2,0.68,0.18,1.08);
  animation: animation-0 1s 0.5s infinite cubic-bezier(0.2,0.68,0.18,1.08);
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

<Loader
  color="#000000"
  css=""
  height={35}
  loading={true}
  margin={2}
  radius={2}
  width={4}
>
  <div
    className="emotion-5"
  >
    <div
      className="emotion-0"
    />
    <div
      className="emotion-1"
    />
    <div
      className="emotion-2"
    />
    <div
      className="emotion-3"
    />
    <div
      className="emotion-4"
    />
  </div>
</Loader>
`;
